R/write.R

```diff
@@ -58,3 +84,36 @@ output_column.POSIXt <- function(x) {
   format(x, "%Y-%m-%dT%H:%M:%OSZ", tz = "UTC")
 }
 
+#' Write a single R object to file
+#'
+#' Consistent wrapper around \code{\link{saveRDS}}. \code{write_rds} never
+#' compresses by default as space is generally cheaper than time.
+#'
+#' @section guess compression:
+#' N.B. this is not yet implemented but some empirical testing may help come up
+#' with something useful.
+#' @param x R object to write to serialise.
+#' @param path Path to write to.
+#' @param compress Compression method to use one of \code{c("none", "gz" ,"bz",
+#' "xz")}.
+#' @param ... Additional arguments to connection function.
+#' @name write_rds
+NULL
+
+#' @rdname write_rds
+#' @export
+#' @examples
+#' ## write_rds(mtcars, "mtcars.rds")
+#' ## write_rds(mtcars, "compressed_mtc.rds", "xz", compression = 9L)
+write_rds <- function(x, path, compress = c("none", "gz", "bz2", "xz"), ...) {
+
+  compress <- match.arg(compress)
+  switch(compress,
+         none = con <-    file(path, ...),
+         gz   = con <-  gzfile(path, ...),
+         bz2  = con <-  bzfile(path, ...),
+         xz   = con <-  xzfile(path, ...))
+
+  saveRDS(x, con)
+  on.exit(close(con))
```

Comment:
The `on.exit()` should go above `saveRDS()` in case `saveRDS` throws an error. It's also a good habit to always do `on.exit(..., add = TRUE)`, so if you have multiple `on.exit()`s they don't override each other